File: 자바/모든문제/_0218/src/BuyCookies.java
public class BuyCookies {
    public int solution(int[] cookie) {
        int answer = 0;

        for (int i = 0; i < cookie.length - 1; i++) {
            int left = i;
            int leftSum = cookie[i];

            int right = i + 1;
            int rightSum = cookie[i + 1];

            while (true) {
                if (leftSum == rightSum && answer < leftSum) {
                    answer = leftSum;
                } else if (leftSum <= rightSum && left != 0) {
                 leftSum += cookie[--left];
                } else if (leftSum > rightSum && right != cookie.length - 1) {
                    rightSum += cookie[++right];
                } else {
                    break;
                }
            }
        }
        return answer;
    }
}

File: 자바스크립트/모든 문제/Day45/Buy Cookies.js
function solution(cookie) {
  let len = cookie.length;
  let answer = 0;

  for (let i = 0; i < len - 1; i++) {
    let left = i;
    let right = i + 1;
    let lsum = cookie[left];
    let rsum = cookie[right];

    while (true) {
      if (lsum === rsum && answer < lsum) {
        answer = lsum;
      } else if (lsum <= rsum && left !== 0) {
        lsum += cookie[--left];
      } else if (rsum <= lsum && right !== len - 1) {
        rsum += cookie[++right];
      } else {
        break;
      }
    }
  }

  return answer;
}
